adapters/ust-ist.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+import { default as moment } from 'moment-timezone';
+import { acceptableParameters, promiseRequest } from '../lib/utils';
+
+export const name = 'ust-ist';
+
+export async function fetchData (source, cb) {
+  try {
+    const allData = JSON.parse(await promiseRequest(source.url));
+    const allMeta = JSON.parse(await promiseRequest('https://api.ust.is/aq/a/getStations'));
+
+    // Generate an array of station IDs there is data for.
+    const stations = Object.keys(allData);
+
+    const measurements = stations.reduce((acc, stationId) => {
+      const stationData = allData[stationId];
+      const stationMeta = allMeta.find(s => s.local_id === stationData.local_id);
+
+      const baseMeta = {
+        location: stationData.name,
+        city: stationMeta.municipality,
+        coordinates: {
+          latitude: Number(stationMeta.latitude),
+          longitude: Number(stationMeta.longitude)
+        }
+      };
+      const latestMeasurements = parseParams(stationData.parameters);
+
+      return acc.concat(latestMeasurements.map(m => ({ ...baseMeta, ...m })));
+    }, []);
+
+    cb(null, {name: 'unused', measurements});
+  } catch (e) {
+    cb(e);
+  }
+}
+
+/**
+ * Parse object with parameters, each with a series of measurements.
+ * Return an array with the latest measurement for valid parameters only.
+ *
+ * @param {object} params Parameter object
+ *
+ * @example parseParams({
+ *   'NO': {
+ *     '0': { endtime: '2020-01-03 03:00:00', value: '0.2175', verification: 3 },
+ *     '1': { endtime: '2020-01-03 02:00:00', value: '0.25', verification: 3 },
+ *     'unit': 'µg/m3',
+ *     'resolution': '1h'
+ *   },
+ *   'NO2': {
+ *     '0': { endtime: '2020-01-03 03:00:00', value: '0.154717', verification: 3 },
+ *     '1': { endtime: '2020-01-03 02:00:00', value: '0.13522', verification: 3 },
+ *     'unit': 'µg/m3',
+ *     'resolution': '1h'
+ *   }
+ * })
+ *
+ * @returns [ { value: 0.154717, date: 2020-01-03 03:00:00. parameter: 'no2' }]
+ *
+ */
+
+function parseParams (params) {
+  // Array with the valid parameters in the object
+  const validParams = Object.keys(params).filter(p => acceptableParameters.includes(p.toLowerCase()));
```

Comment:
This is leaving out PM2.5 since the acceptable parameter looks like 'pm25' and this makes it 'pm2.5'

Reply:
Fixed in the latest commit.